app/models/manageiq/providers/vmware/infra_manager/refresher.rb

```diff
@@ -18,132 +19,53 @@ def provider.use_vim_broker?; @__use_vim_broker; end
-        start_time = Time.now
-
-        until targets_with_data.empty?
-          target, data = targets_with_data.shift
-
-          _log.info "#{log_header} Refreshing target #{target.class} [#{target.name}] id [#{target.id}]..."
-
-          hashes = parse_data(data)
-
-          # We no longer need the filtered VC data, so remove it to help the GC
-          data = nil
-
-          save_target(target, hashes)
-
-          _log.info "#{log_header} Refreshing target #{target.class} [#{target.name}] id [#{target.id}]...Complete"
-        end
-
-        Benchmark.realtime_block(:post_refresh_ems) { post_refresh_ems(start_time) }
-      end
-
-      def get_and_filter_vc_data(targets)
-        Benchmark.realtime_block(:get_ems_data) { get_ems_data }
-        Benchmark.realtime_block(:get_vc_data) { get_vc_data }
-
-        Benchmark.realtime_block(:get_vc_data_ems_customization_specs) { get_vc_data_ems_customization_specs } if targets.include?(@ems)
+        Benchmark.realtime_block(:get_vc_data_ems_customization_specs) { get_vc_data_ems_customization_specs(ems) } if targets.include?(ems)
 
         # Filter the data, and determine for which hosts we will need to get extended data
         filtered_host_mors = []
         targets_with_data = targets.collect do |target|
-          filtered_data, = Benchmark.realtime_block(:filter_vc_data) { filter_vc_data(target) }
+          filtered_data, = Benchmark.realtime_block(:filter_vc_data) { filter_vc_data(ems, target) }
           filtered_host_mors += filtered_data[:host].keys
           [target, filtered_data]
         end
         filtered_host_mors.uniq!
 
-        Benchmark.realtime_block(:get_vc_data_host_scsi) { get_vc_data_host_scsi(filtered_host_mors) }
+        Benchmark.realtime_block(:get_vc_data_host_scsi) { get_vc_data_host_scsi(ems, filtered_host_mors) }
 
         return targets_with_data
       ensure
-        disconnect_from_ems
+        disconnect_from_ems(ems)
       end
 
-      def parse_data(data)
-        log_header = "EMS: [#{@ems.name}], id: [#{@ems.id}]"
+      def parse_targeted_inventory(ems, inventory, _targets)
+        log_header = "EMS: [#{ems.name}], id: [#{ems.id}]"
         _log.debug "#{log_header} Parsing VC inventory..."
         hashes, = Benchmark.realtime_block(:parse_vc_data) do
-          InfraManager::RefreshParser.ems_inv_to_hashes(data)
+          InfraManager::RefreshParser.ems_inv_to_hashes(inventory)
         end
         _log.debug "#{log_header} Parsing VC inventory...Complete"
 
         hashes
       end
 
-      def save_target(target, hashes)
+      def save_inventory(ems, target, hashes)
         Benchmark.realtime_block(:db_save_inventory) do
-          @ems.update_attributes(@ems_data) unless @ems_data.nil?
-          EmsRefresh.save_ems_inventory(@ems, hashes, target)
+          # TODO: really wanna kill this @ems_data instance var
+          ems.update_attributes(@ems_data) unless @ems_data.nil?
```

Comment:
Some use the `@ems_data` ... specifically the ones that update the `api_version` for the provider.  I left this where it's at for now only because it wasn't consistently used.